### PR TITLE
feat: add `Bernstein` polynomial PDF

### PR DESCRIPTION
**Pull request by ikrommyd**
Adds `Bernstein` polynomial PDF constructed as a linear combination of Bernstein polynomials
Related to #512 

## Proposed Changes

- Implement `Bernstein` polynomial PDF

## Tests added

- Modify `tests/serialize/test_basics.py`  and `tests/test_pdf_polynomials.py`

## Checklist

- [x] change approved
- [x] implementation finished
- [x]  docs updated
- [x] correct namespace imported
- [x] tests added
- [x] CHANGELOG updated
- [x] (if new public method/class) docs in rst file updated?
- [x] (if large change) tutorial added/updated?

### CODE REVIEW

**Comment by ikrommyd** [comment]
@jonas-eschle Could you please check the `BernsteinPDFRepr` for what I'm doing wrong there? It gives `yaml.representer.RepresenterError: ('cannot represent an object', <zfit.<class 'zfit.models.polynomials.Bernstein'>  params=[coeff0_bernstein, coeff1_bernstein, coeff2_bernstein, coeff3_bernstein, coeff4_bernstein])
` when recreating truths

**Comment by ikrommyd** [comment]
> @jonas-eschle Could you please check the `BernsteinPDFRepr` for what I'm doing wrong there? It gives `yaml.representer.RepresenterError: ('cannot represent an object', <zfit.<class 'zfit.models.polynomials.Bernstein'> params=[coeff0_bernstein, coeff1_bernstein, coeff2_bernstein, coeff3_bernstein, coeff4_bernstein]) ` when recreating truths

nvm I think I got it

**Comment by ikrommyd** [comment]
@jonas-eschle I'm happy with the current state. Please review

**Review by jonas-eschle — APPROVED** [review]
Looks good! One minor suggestion, but that should not change anything, apart of that, lgtm!

File: zfit/models/polynomials.py [hunk 1]
```diff
@@ -789,6 +789,150 @@ def indefinite_integral(limits):
...
+            extended: |@doc:pdf.init.extended| The overall yield of the PDF.
+               If this is parameter-like, it will be used as the yield,
+               the expected number of events, and the PDF will be extended.
+               An extended PDF has additional functionality, such as the
+               ``ext_*`` methods and the ``counts`` (for binned PDFs). |@docend:pdf.init.extended|
+            norm: |@doc:pdf.init.norm| Normalization of the PDF.
+               By default, this is the same as the default space of the PDF. |@docend:pdf.init.norm|
+            name: Name of the polynomial
+        """
+        coeffs = convert_to_container(coeffs).copy()
```
[the remark this answers is not in the document]
Comment (jonas-eschle) [reply zfit/models/polynomials.py:853]: let me know if you agree and commit, then I'll merge

[the remark this answers is not in the document]
Comment (ikrommyd) [reply zfit/models/polynomials.py:853]: should I also remove it from the `RecursivePolynomial` class?

[the remark this answers is not in the document]
Comment (ikrommyd) [reply zfit/models/polynomials.py:853]: ah no there we re modifying the content

**Pull request merged**